src/main/java/Parser.java

```diff
@@ -0,0 +1,37 @@
+public class Parser {
+    private String[] parsedInput;
+
+    public Parser(String line) throws InvalidCommandException, MissingDescriptionException {
+        parsedInput = line.split(" ");
+
+        switch (parsedInput[0]) {
+        case "todo":
+            if (parsedInput.length < 2) {
+                throw new MissingDescriptionException();
+            }
+            break;
+        case "deadline":
+            // Fallthrough
+        case "event":
+            // Fallthrough
+        case "list":
+            break;
+        case "delete":
+            // Fallthrough
+        case "mark":
+            // Fallthrough
+        case "unmark":
+            if (parsedInput.length != 2) { //simple workaround for now, need better checks later
```

Comment:
It would be better to insert the [comment](https://se-education.org/guides/conventions/java/basic.html#comments) in a separate line, with same level of indentation as the code. This will not break the logical structure of the code.